ws/ws.go

```diff
@@ -102,7 +103,8 @@ func Handle(w http.ResponseWriter, r *http.Request) {
 	// Advance deadline if we receive a pong.
 	// TODO(flowlo): maybe check whether the contents of the
 	// message actually match those of a pending ping.
-	ws.SetPongHandler(func(_ string) error {
+	ws.SetPongHandler(func(s string) error {
+		log.Debugf(appengine.BackgroundContext(), "ws: pong %x for %s(%d %q) sent", s, id.kind, id.intID, id.stringID)
```

Comment:
This is just plain wrong. Pongs are only sent in response to pings. The pong handler is there in order to digest incoming pongs and match them against previously requested pings. E.g. this must log the reception of a pong, not the  transmission.